File: P74 Search a 2D Matrix/solution.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m = len(matrix)
        if m == 0:
            return False
        n = len(matrix[0])
        if n == 0:
            return False
        i = 0
        while i < m:
            if target < matrix[i][0]:
                return False
            elif target <= matrix[i][n-1]:
                for j in range(len(matrix[i])):
                    if target == matrix[i][j]:
                        return True
                break
            else:
                i += 1
        return False

File: P74 Search a 2D Matrix/Solution.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m = matrix.length;
        if(m == 0) {
            return false;
        }
        int n = matrix[0].length;
        if(n == 0) {
            return false;
        }
        int i = 0;
        while(i < m) {
            if(target < matrix[i][0]) {
                return false;
            } else if(target <= matrix[i][n-1]) {
                for(int j = 0; j < matrix[i].length; j++) {
                    if(matrix[i][j] == target) {
                        return true;
                    }
                }
                break;
            } else {
                i++;
            }
        }
        return false;
    }
}